http-generator-core/src/main/java/io/avaje/http/generator/core/ElementReader.java

```diff
@@ -56,6 +70,46 @@ public class ElementReader {
     }
   }
 
+  TypeHandler initTypeHandler() {
+
+    if (specialParam) {
+
+      final var typeOp =
+          Optional.ofNullable(type).or(() -> Optional.of(UType.parse(element.asType())));
+
+      final var mainTypeEnum =
+          typeOp
+              .flatMap(t -> Optional.ofNullable(ctx.typeElement(t.mainType())))
+              .map(TypeElement::getKind)
+              .filter(ElementKind.ENUM::equals)
+              .isPresent();
+
+      final var isCollection =
+          typeOp
+              .filter(t -> t.isGeneric() && !t.mainType().startsWith("java.util.Map"))
```

Comment:
`Optional<>` params are gonna haunt you with this.

Reply:
Also, what if my BeanParam has a Map<String, String> field in it? Retrofit2 supports all of this stuff and can be deserialized by general json deserialization as any "BeanParam" would be treated as a Body request (GET/POST/PATCH/PUT). Could we just do the same?

Reply:
Oh we already support maps

Reply:
> `Optional<>` params are gonna haunt you with this.

Well, it's a good thing we don't support `Optional<>` then.